### PR TITLE
Feature/apm user

### PR DESCRIPTION
**Pull request by Shahar-Y**
Added user id to apm transaction.
Tested with apm.

### CODE REVIEW

File: server/auth/auth.go [hunk 1]
```diff
@@ -71,6 +72,10 @@ func (r *Router) Middleware(secret string, authURL string) gin.HandlerFunc {
 			return
 		}
 
+		// ct is the current transaction of the apm. adding the user id to the context.
+		ct := apm.TransactionFromContext(c.Request.Context())
```
Comment (omerzamir) [inline server/auth/auth.go]: Bad naming...

Comment (Shahar-Y) [reply server/auth/auth.go]: I thought that's how golang goes.. 
Any suggestion?

Comment (omerzamir) [reply server/auth/auth.go]: No it doesn't (you only name receivers like this, names should be indicative).
How about `currentTransaction` , just being creative...

File: server/auth/auth.go [hunk 2]
```diff
@@ -71,6 +72,10 @@ func (r *Router) Middleware(secret string, authURL string) gin.HandlerFunc {
...
+		ct.Context.SetUserID(id)
```
Comment (omrishtam) [inline server/auth/auth.go]: [this](https://www.elastic.co/guide/en/apm/agent/go/current/api.html#context-set-username) exists too.

Comment (Shahar-Y) [reply server/auth/auth.go]: True. but SetUsername doesn't work. SetUserEmail does. 
Decided on not implementing it

File: server/auth/auth.go (hunk 1 above)
Comment (Shahar-Y) [reply server/auth/auth.go]: Done.

**Pull request merged**